subprojects/platform-jvm/src/main/java/org/gradle/platform/internal/DefaultBuildPlatform.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.platform.internal;
+
+import net.rubygrapefruit.platform.SystemInfo;
+import org.gradle.api.GradleException;
+import org.gradle.platform.Architecture;
+import org.gradle.platform.BuildPlatform;
+import org.gradle.platform.OperatingSystem;
+
+import javax.inject.Inject;
+
+public class DefaultBuildPlatform implements BuildPlatform {
+
+    private final SystemInfo systemInfo;
+
+    private final org.gradle.internal.os.OperatingSystem operatingSystem;
+
+    @Inject
+    public DefaultBuildPlatform(SystemInfo systemInfo, org.gradle.internal.os.OperatingSystem operatingSystem) {
+        this.systemInfo = systemInfo;
```

Comment:
Unless there is a lazy consideration, I would do the mapping in the constructor once.